docs/3.api/4.kit/10.templates.md

```diff
@@ -0,0 +1,281 @@
+---
+title: "Templates"
+description: Nuxt Kit provides a set of utilities to help you work with templates. These functions allow you to generate extra files during development and build time.
+---
+
+# Templates
+
+[source code](https://github.com/nuxt/nuxt/blob/main/packages/kit/src/template.ts)
+
+Templates allows to generate extra files during development and build time. These files will be available in virtual filesystem and can be used in plugins, layouts, components, etc. `addTemplate` and `addTypeTemplate` allow you to add templates to the Nuxt application. `updateTemplates` allows you to regenerate templates that match the filter.
+
+## `addTemplate`
+
+Renders given template using [lodash template](https://lodash.com/docs/4.17.15#template) during build into the project buildDir.
```

Comment:
Agreed, I saw the TODO mentioning it, only thought it might be helpful for people to find extra documentation. Of course we can remove it in purpose of future deprecations.